### DISCUSSION TITLE
Sending Authorization in url

### DISCUSSION THREAD

**Discussion by DorSA9** [netbox-community/netbox#6993] [Q&A]
Hello,
I have netbox with the version of 2.9.8.
I have a load balancer (F5), which need to connect to IPAM for getting an ip address for a VIP.
I have a problem that the F5 only support connection to an api with basic auth, and not with a token.
I tried to push the authorization token into the url : **POST  <netbox_ip>/api?Authorization?: Token <TOKEN>/ipam/prefixes/id/available-ips/**
but when  i do that im getting an response 301 Moved Permanently and the it redirect it into a **GET <netbox_ip>/api?Authorization?: Token <TOKEN>/ipam/prefixes/id/available-ips/**

and I didnt get any ip address but the following response:

{"circuits":"http://NETBOX_IP/api/circuits/","dcim":"http://NETBOX_IP/api/dcim/","extras":"http://NETBOX_IP/api/extras/","ipam":"http://NETBOX_IP/api/ipam/","plugins":"http://NETBOX_IP/api/plugins/","secrets":"http://NETBOX_IP/api/secrets/","tenancy":"http://NETBOX_IP/api/tenancy/","users":"http://NETBOX_IP/api/users/","virtualization":"http://NETBOX_IP/api/virtualization/"}

I can also send a pcap file so you can see the whole traffic

**Comment by sdktr** [comment]
Netbox doesn't support authorization in the URL. Basic Auth uses a header as well. The output you got is from the user friendly /api page which renders in a browser if you do not supply the correct Content-Type.

301 permanent redirect might be because you missed a required trailing slash after your request?

**Comment by DorSA9** [comment]
Is there any option to support with basic auth instead of token?
basic auth is the only option that I can to with The BIGIP product

‫בתאריך יום ו׳, 20 באוג׳ 2021 ב-17:13 מאת ‪Stefan de Kooter‬‏ <‪
***@***.***‬‏>:‬

**Reply by sdktr** [reply]
That could be a job for the webserver or frontend proxy that's in front of your netbox. If that is running nginx, it could handle the basic auth: https://docs.nginx.com/nginx/admin-guide/security-controls/configuring-http-basic-authentication/
 Add a netbox api token to the headers that nginx uses to connect to the netbox app server: https://stackoverflow.com/a/32834999
